Build with Java 21

**Is your feature request related to a problem? Please describe.**
When building with Java 21 you get:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project intersmash-kubernetes-client: Fatal error compiling: java.lang.NoSuchFieldError: Class com.sun.tools.javac.tree.JCTree$JCImport does not have member field 'com.sun.tools.javac.tree.JCTree qualid' -> [Help 1]
```

**Describe the solution you'd like**
Make it build with Java 21

--- a/.github/workflows/checkstyle.yml
+++ b/.github/workflows/checkstyle.yml
@@ -22,7 +22,5 @@ jobs:
           java-version: '17'
           distribution: 'temurin'
           cache: 'maven'
-      - name: Pre-fetch required dependencies
-        run: mvn clean install --batch-mode -pl style-config -am -DskipTests
       - name: Check source code formatting, file headers, etc.
-        run: mvn spotless:check --batch-mode
+        run: mvn -T 1C spotless:check --batch-mode -Daether.dependencyCollector.impl=bf

--- a/.github/workflows/kubernetes-e2e.yml
+++ b/.github/workflows/kubernetes-e2e.yml
@@ -17,13 +17,14 @@ concurrency:
 
 jobs:
   minikube:
-    name: K8S
+    name: "K8S Integration tests on K8S: ${{ matrix.kubernetes }} with JDK: ${{ matrix.java-version }}"
     runs-on: ubuntu-latest
     if: github.repository == 'Intersmash/intersmash'
     strategy:
       fail-fast: false
       matrix:
-        kubernetes: [v1.33.1]
+        kubernetes: [v1.35.2]
+        java-version: ['17','21']
     steps:
       - name: Checkout
         uses: actions/checkout@v6
@@ -32,7 +33,7 @@ jobs:
         with:
           driver: docker
           minikube version: 'v1.38.1'
-          kubernetes version: 'v1.35.2'
+          kubernetes version: ${{ matrix.kubernetes }}
           github token: ${{ secrets.GITHUB_TOKEN }}
           start args: "--memory='4gb' --cpus='2'"
       - name: Install operator-sdk
@@ -47,20 +48,23 @@ jobs:
           curl -LO ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH}
           chmod +x operator-sdk_${OS}_${ARCH} && sudo mv operator-sdk_${OS}_${ARCH} /usr/local/bin/operator-sdk
           operator-sdk olm install
-      - name: Set up JDK 17
+      - name: Mirror hyperfoil kube-rbac-proxy image needed by Hyperfoil
+        run: minikube ssh "docker pull registry.k8s.io/kubebuilder/kube-rbac-proxy:v0.15.0 && docker tag registry.k8s.io/kubebuilder/kube-rbac-proxy:v0.15.0 gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0"
+      - name: Set up JDK ${{ matrix.java-version }}
         uses: actions/setup-java@v5
         with:
-          java-version: '17'
+          java-version: ${{ matrix.java-version }}
           distribution: 'temurin'
           cache: 'maven'
-      - name: Install and Run Integration Tests
+      - name: Prepare local Maven repository
         run: |
           set -x
-          #
           mkdir local-repo
-          # pre-fetch dependencies
-          mvn clean install -pl kubernetes-client,core,provisioners,testsuite,testsuite/deployments,testsuite/deployments/deployments-provider -am --batch-mode -Dmaven.repo.local=./local-repo -DskipTests
-          # run K8s integration tests 
+          set +x
+      - name: Pre-fetch dependencies
+        run: mvn clean install -T 1C -pl kubernetes-client,core,provisioners,testsuite,testsuite/deployments,testsuite/deployments/deployments-provider -am --batch-mode -Dmaven.repo.local=./local-repo -DskipTests -Daether.dependencyCollector.impl=bf
+      - name: Run Integration Tests
+        run: |
           mvn test -e --batch-mode -fae -Dmaven.repo.local=./local-repo -pl testsuite/integration-tests -Pts.k8s \
           -Dintersmash.kubernetes.url=https://$(minikube ip):8443 \
           -Dintersmash.kubernetes.route_domain=$(minikube ip) \

--- a/.github/workflows/simple-build.yml
+++ b/.github/workflows/simple-build.yml
@@ -10,15 +10,24 @@ jobs:
   simple-build:
     if: '! github.event.pull_request.draft'
     runs-on: ubuntu-latest
-
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version: ['17', '21']
     steps:
       - name: Checkout
         uses: actions/checkout@v6
-      - name: Set up JDK 17
+      - name: Set up JDK ${{ matrix.java-version }}
         uses: actions/setup-java@v5
         with:
-          java-version: '17'
+          java-version: ${{ matrix.java-version }}
           distribution: 'temurin'
           cache: 'maven'
-      - name: Build with Maven
-        run:  mvn -version ; mvn clean install -U -B -DskipTests ; mvn test -B -pl testsuite/deployments/deployments-provider,kubernetes-client,core,provisioners -am
+      - name: Display environment info
+        run: |
+          mvn -version
+          java -version
+      - name: Build repository
+        run: mvn clean install -T 1C -U -B -DskipTests -Daether.dependencyCollector.impl=bf
+      - name: Run smoke tests
+        run: mvn test -B -pl testsuite/deployments/deployments-provider,kubernetes-client,core,provisioners

--- a/.gitignore
+++ b/.gitignore
@@ -47,3 +47,12 @@ infra/maven-cache/indy
 
 # Ansible
 *.retry
+
+# VSCode
+.vscode
+
+# Misc
+.java-version
+
+# Claude
+CLAUDE.md

--- a/kubernetes-client/src/main/java/org/jboss/intersmash/k8s/client/binary/ClusterVersionBasedKubernetesClientBinaryPathResolver.java
+++ b/kubernetes-client/src/main/java/org/jboss/intersmash/k8s/client/binary/ClusterVersionBasedKubernetesClientBinaryPathResolver.java
@@ -43,7 +43,7 @@ import lombok.extern.slf4j.Slf4j;
  */
 @Slf4j
 public class ClusterVersionBasedKubernetesClientBinaryPathResolver implements KubernetesClientBinaryPathResolver {
-	private static final String KUBERNETES_CLIENT_BINARY_DOWNLOAD_BASE_URL = "https://dl.k8s.io/release";
+	private static final String KUBERNETES_CLIENT_BINARY_DOWNLOAD_BASE_URL = "https://dl.k8s.io";
 	public static final int BINARY_DOWNLOAD_CONNECTION_TIMEOUT = 20_000;
 	public static final int BINARY_DOWNLOAD_READ_TIMEOUT = 300_000;
 
@@ -92,6 +92,9 @@ public class ClusterVersionBasedKubernetesClientBinaryPathResolver implements Ku
 		String arch = "amd64";
 		if (SystemUtils.IS_OS_MAC) {
 			systemType = "darwin";
+			if (!"amd64".equals(SystemUtils.OS_ARCH)) {
+				arch = "arm64";
+			}
 		}
 		return String.format("%s/%s/bin/%s/%s/%s", KUBERNETES_CLIENT_BINARY_DOWNLOAD_BASE_URL, clusterVersion.getGitVersion(),
 				systemType, arch, BINARY_NAME);

--- a/pom.xml
+++ b/pom.xml
@@ -45,8 +45,8 @@
     <properties>
         <minimum.maven.version>3.3.9</minimum.maven.version>
 
-        <maven.compiler.source>11</maven.compiler.source>
-        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
 
         <xtf.version>1.0</xtf.version>
 
@@ -68,8 +68,7 @@
             Note: we are using a custom index image built as described in https://github.com/Intersmash/intersmash/issues/32
         -->
         <version.intersmash.activemq.operators>v1.0.11</version.intersmash.activemq.operators>
-
-        <version.maven-compiler-plugin>3.8.1</version.maven-compiler-plugin>
+        <version.maven-compiler-plugin>3.13.0</version.maven-compiler-plugin>
         <version.maven-checkstyle-plugin>3.1.1</version.maven-checkstyle-plugin>
         <version.maven-clean-plugin>3.1.0</version.maven-clean-plugin>
         <version.maven-enforcer-plugin>3.0.0</version.maven-enforcer-plugin>
@@ -156,6 +155,13 @@
                 <groupId>org.projectlombok</groupId>
                 <artifactId>lombok</artifactId>
                 <version>${version.lombok}</version>
+                <scope>provided</scope>
+            </dependency>
+            <dependency>
+                <groupId>io.sundr</groupId>
+                <artifactId>builder-annotations</artifactId>
+                <version>${version.sundrio.builder-annotations}</version>
+                <scope>provided</scope>
             </dependency>
             <dependency>
                 <groupId>ch.qos.logback</groupId>
@@ -407,13 +413,6 @@
                 <artifactId>kubernetes-client-api</artifactId>
                 <version>${version.io.fabric8}</version>
             </dependency>
-
-            <dependency>
-                <groupId>io.sundr</groupId>
-                <artifactId>builder-annotations</artifactId>
-                <version>${version.sundrio.builder-annotations}</version>
-            </dependency>
-
             <dependency>
                 <groupId>org.eclipse.aether</groupId>
                 <artifactId>aether-api</artifactId>
@@ -481,17 +480,18 @@
                     <version>${version.maven-compiler-plugin}</version>
                     <configuration>
                         <annotationProcessorPaths>
-                            <path>
-                                <groupId>org.projectlombok</groupId>
-                                <artifactId>lombok</artifactId>
-                                <version>${version.lombok}</version>
-                            </path>
-                            <path>
-                                <groupId>io.sundr</groupId>
-                                <artifactId>builder-annotations</artifactId>
-                                <version>${version.sundrio.builder-annotations}</version>
-                            </path>
+                          <path>
+                            <groupId>io.sundr</groupId>
+                            <artifactId>builder-annotations</artifactId>
+                            <version>${version.sundr.builder-annotations}</version>
+                          </path>
+                          <path>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok</artifactId>
+                            <version>${version.lombok}</version>
+                          </path>
                         </annotationProcessorPaths>
+                        <useIncrementalCompilation>false</useIncrementalCompilation>
                     </configuration>
                 </plugin>
                 <plugin>
@@ -573,6 +573,13 @@
                     <groupId>com.diffplug.spotless</groupId>
                     <artifactId>spotless-maven-plugin</artifactId>
                     <version>${version.spotless-maven-plugin}</version>
+                    <configuration>
+                        <java>
+                          <eclipse>
+                            <file>${maven.multiModuleProjectDirectory}/style-config/src/main/resources/org/jboss/intersmash/configs/eclipse-formatter.xml</file>
+                          </eclipse>
+                        </java>
+                    </configuration>
                 </plugin>
                 <plugin>
                     <groupId>io.fabric8</groupId>
@@ -629,22 +636,12 @@
             <plugin>
                 <groupId>com.diffplug.spotless</groupId>
                 <artifactId>spotless-maven-plugin</artifactId>
-                <dependencies>
-                    <dependency>
-                        <groupId>org.jboss.intersmash</groupId>
-                        <artifactId>style-config</artifactId>
-                        <version>${project.version}</version>
-                    </dependency>
-                </dependencies>
                 <configuration>
                     <java>
                         <removeUnusedImports />
                         <importOrder>
                             <order>jakarta,java,javax,org,com</order>
                         </importOrder>
-                        <eclipse>
-                            <file>org/jboss/intersmash/configs/eclipse-formatter.xml</file>
-                        </eclipse>
                         <licenseHeader>
                             <content>
 /*

--- a/testsuite/deployments/eap7-shared/pom.xml
+++ b/testsuite/deployments/eap7-shared/pom.xml
@@ -63,6 +63,11 @@
                                 <location>${eap7.xp-feature-pack.location}</location>
                             </feature-pack>
                         </feature-packs>
+                        <plugin-options>
+                            <!-- this flag is super necessary: otherwise wildfly-maven-plugin and
+                                 wildfly-jar-maven-plugin would collide  -->
+                            <jboss-fork-embedded>true</jboss-fork-embedded>
+                        </plugin-options>
                     </configuration>
                 </plugin>
             </plugins>

--- a/testsuite/deployments/openshift-jakarta-sample-standalone/pom.xml
+++ b/testsuite/deployments/openshift-jakarta-sample-standalone/pom.xml
@@ -10,8 +10,8 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.source>11</maven.compiler.source>
-        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
 
         <version.wildfly-server>38.0.0.Beta1</version.wildfly-server>
         <!-- WildFly Maven Plugin coordinates -->
@@ -47,7 +47,6 @@
         <wildfly.xp-channel.version />
 
         <maven-war-plugin.version>3.3.2</maven-war-plugin.version>
-        <version.spotless-maven-plugin>2.35.0</version.spotless-maven-plugin>
         <version.maven-deploy-plugin>3.0.0-M1</version.maven-deploy-plugin>
     </properties>
 
@@ -105,7 +104,6 @@
                 <plugin>
                     <groupId>com.diffplug.spotless</groupId>
                     <artifactId>spotless-maven-plugin</artifactId>
-                    <version>${version.spotless-maven-plugin}</version>
                 </plugin>
             </plugins>
         </pluginManagement>
@@ -142,22 +140,12 @@
             <plugin>
                 <groupId>com.diffplug.spotless</groupId>
                 <artifactId>spotless-maven-plugin</artifactId>
-                <dependencies>
-                    <dependency>
-                        <groupId>org.jboss.intersmash</groupId>
-                        <artifactId>style-config</artifactId>
-                        <version>0.0.5.Beta1-SNAPSHOT</version>
-                    </dependency>
-                </dependencies>
                 <configuration>
                     <java>
                         <removeUnusedImports />
                         <importOrder>
                             <order>jakarta,java,javax,org,com</order>
                         </importOrder>
-                        <eclipse>
-                            <file>org/jboss/intersmash/configs/eclipse-formatter.xml</file>
-                        </eclipse>
                         <licenseHeader>
                             <content>
 /*

--- a/testsuite/deployments/wildfly-shared/pom.xml
+++ b/testsuite/deployments/wildfly-shared/pom.xml
@@ -130,6 +130,11 @@
                                 <location>${wildfly.feature-pack.location}</location>
                             </feature-pack>
                         </feature-packs>
+                        <plugin-options>
+                            <!-- this flag is super necessary: otherwise wildfly-maven-plugin and
+                                 wildfly-jar-maven-plugin would collide  -->
+                            <jboss-fork-embedded>true</jboss-fork-embedded>
+                        </plugin-options>
                     </configuration>
                 </plugin>
             </plugins>

--- a/testsuite/integration-tests/src/test/java/org/jboss/intersmash/testsuite/provision/k8s/HyperfoilKubernetesOperatorProvisionerTest.java
+++ b/testsuite/integration-tests/src/test/java/org/jboss/intersmash/testsuite/provision/k8s/HyperfoilKubernetesOperatorProvisionerTest.java
@@ -30,7 +30,6 @@ import org.jboss.intersmash.testsuite.k8s.NamespaceCreationCapable;
 import org.jboss.intersmash.testsuite.provision.operator.HyperfoilOperatorProvisionerTests;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.MethodOrderer;
 import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
@@ -39,7 +38,6 @@ import org.junit.jupiter.api.TestMethodOrder;
 import io.hyperfoil.v1alpha2.Hyperfoil;
 import io.hyperfoil.v1alpha2.HyperfoilBuilder;
 
-@Disabled("suspended because fails on PRs testing")
 @TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 @KubernetesTest
 public class HyperfoilKubernetesOperatorProvisionerTest implements NamespaceCreationCapable {

